nengo/builder/connection.py

```diff
@@ -219,8 +219,11 @@ def get_prepost_signal(is_pre):
             targets.append(r.modifies)
 
         if 'encoders' in targets:
-            model.sig[conn.post_obj]['encoders'].readonly = False
-            model.add_op(PreserveValue(model.sig[conn.post_obj]['encoders']))
+            encoder_sig = model.sig[conn.post_obj]['encoders']
+            if not any(isinstance(op, PreserveValue) and op.dst is encoder_sig
+                       for op in model.operators):
```

Comment:
It's overkill. Even for a huge model, this is completely negligible compared with solving for decoders and things like that.

But having better storage of operators is something I've wanted in the past, too, particularly when I was looking at optimizers. Do you think it would make sense to store them based on the object that creates them, like we do with signals? That's what I had thought in the past, and I think it would work fairly well. It's not quite as clear-cut as with signals, since many operators copy things from signals belonging to one object to signals belonging to another object. But each operator is unambiguously created in some object's builder, so I think it makes sense to assign ownership that way. I'll make an issue and note that this is one place it would help.